packages/sync-actions/src/utils/extract-matching-pairs.js

```diff
@@ -0,0 +1,16 @@
+export default function extractMatchingPairs(hashMap, key, before, now) {
```

Comment:
shall we use named arguments such as `({ path, key, previous, next })`?
I've inspected `hashMap` and it really isn't a hashMap at all. what it is a key value object where the value is actually the path to the specified changes. the key it is indication of change or remove